qiskit_metal/renderers/renderer_gmsh/gmsh_renderer.py

```diff
@@ -0,0 +1,1129 @@
+                self.design.ls.get_properties_for_layer_datatype(
+                    properties=props, layer_number=qc_layer))
+            if result:
+                qc_thickness, qc_z = result
+            else:
+                raise ValueError(
+                    f"Could not find {props} for the layer_number={qc_layer} in component"
+                    f" '{qcomp.name}'. Check your design and try again.")
+
+            # if qc_thickness is None or qc_z is None:
+            #     self.logger.error("Properties for")
+            # qc_chip_z = self.parse_units_gmsh(self.design.get_chip_z(chip_name))
+            rect_mid = mid + normal * gap / 2
+            rect_vec = np.array([rect_mid[0], rect_mid[1], qc_z])
+            # Assumption: pins only point in x or y directions
+            # If this assumption is not satisfied, draw_rect_center no longer works -> must use draw_polyline
+            if abs(normal[0]) > abs(normal[1]):
+                dx = gap
+                dy = width + 2 * gap
+                rect_x = rect_vec[0] - dx / 2
+                rect_y = rect_vec[1] - dy / 2
+                rect_z = rect_vec[2]
+            else:
+                dy = gap
+                dx = width + 2 * gap
+                rect_x = rect_vec[0] - dx / 2
+                rect_y = rect_vec[1] - dy / 2
+                rect_z = rect_vec[2]
+
+            # TODO: 3D change: addBox instead of addRectangle
+            if np.abs(qc_thickness) > 0:
+                endcap = gmsh.model.occ.addBox(x=rect_x,
+                                               y=rect_y,
+                                               z=rect_z,
+                                               dx=dx,
+                                               dy=dy,
+                                               dz=qc_thickness)
+            else:
+                endcap = gmsh.model.occ.addRectangle(x=rect_x,
+                                                     y=rect_y,
+                                                     z=rect_z,
+                                                     dx=dx,
+                                                     dy=dy)
+            self.layer_subtract_dict[qc_layer].add(endcap)
+
+    def render_layers(self,
+                      layers: Optional[List[int]] = None,
+                      draw_sample_holder: bool = True,
+                      box_plus_buffer: bool = True):
+        """Render all chips of the design. calls `render_chip` to render the actual geometries
+
+        Args:
+            layers (Optional[List[int]]): List of layers to render.
+                                Renders all if [] or None is given. Defaults to None.
+            draw_sample_holder (bool, optional): To draw the sample holder box. Defaults to True.
+            box_plus_buffer (bool, optional): For adding buffer to chip dimensions. Defaults to True.
+        """
+        # TODO: 3D change: give option to render layers instead of chips?
```

Comment:
Can you please make an issue and give a reference to an issue number?

Reply:
removed the TODO, as the need is already satisfied in the updated code.